src/external/kvn_safe_callback.hpp

```diff
@@ -48,7 +48,7 @@ class safe_callback<_Res(_ArgTypes...)> {
     _Res operator()(_ArgTypes... arguments) {
         if (_is_loaded) {
             std::scoped_lock lock(_mutex);
-            return _callback(arguments...);
+            return _callback(static_cast<_ArgTypes&&>(arguments)...);
```

Comment:
Is this static cast required?

Reply:
This is equivalent to `std::forward<_ArgTypes>`, I use the cast version because it does not require you to include massive standard library header which slows down compile times. It is needed here, because without it you can't really use `kvn::safe_callback` with any callable that takes r-value references as arguments (without the cast, they would be passed to the stored callable as l-values, causing a compilation error).